pkg/i2gw/providers/openapi/utils.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapi
+
+// Map applies the given mapper function to each element in the input slice and returns a new slice with the results.
+func Map[T, U any](slice []T, f func(T) U) []U {
+	arr := make([]U, len(slice))
+	for i, e := range slice {
+		arr[i] = f(e)
+	}
+	return arr
+}
+
+// Filter filters the input slice using the given predicate function and returns a new slice with the results.
+func Filter[T any](slice []T, f func(T) bool) []T {
+	arr := make([]T, 0)
+	for _, e := range slice {
+		if f(e) {
+			arr = append(arr, e)
+		}
```

Comment:
Nice! would that be useful to have these utils shared outside of openapi package?

Reply:
Happy to move them to the `common` package if you say it's worthwhile.

Reply:
Moved the funcs to `provider/common`.